spread.yaml

```
@@ -0,0 +1,149 @@
+          username: spread
+    prepare: |
+      systemctl disable --now unattended-upgrades.service
+      systemctl mask unattended-upgrades.service
+      pipx install charmcraftcache
+      cd "$SPREAD_PATH"
+      bash scripts/build_lib_for_integration.sh
+    restore-each: |
+      cd "$SPREAD_PATH"
+      # Revert python-libjuju version override
+      git restore pyproject.toml poetry.lock
+
+      # Use instead of `concierge restore` to save time between tests
+      # For example, with microk8s, using `concierge restore` takes twice as long as this (e.g. 6
+      # min instead of 3 min between every spread job)
+      juju destroy-model --force --no-wait --destroy-storage --no-prompt concierge-microk8s:testing
+      juju destroy-model --force --no-wait --destroy-storage --no-prompt concierge-lxd:testing
+      juju kill-controller --no-prompt concierge-microk8s
+      juju kill-controller --no-prompt concierge-lxd
+    restore: |
+      rm -rf "$SPREAD_PATH"
+
+  github-ci:
+    type: adhoc
+    # Only run on CI
+    manual: true
+    # HACK: spread requires runners to be accessible via SSH
+    # Configure local sshd & instruct spread to connect to the same machine spread is running on
+    # (spread cannot provision GitHub Actions runners, so we provision a GitHub Actions runner for
+    # each spread job & select a single job when running spread)
+    # Derived from https://github.com/jnsgruk/zinc-k8s-operator/blob/a21eae8399eb3b9df4ddb934b837af25ef831976/spread.yaml#L47
+    allocate: |
+      sudo tee /etc/ssh/sshd_config.d/10-spread-github-ci.conf << 'EOF'
+      PasswordAuthentication yes
+      PermitEmptyPasswords yes
+      EOF
+
+      sudo useradd spread -s /bin/bash -m || true
+      echo "spread:$SPREAD_PASSWORD" | sudo chpasswd || true
+      echo 'spread ALL=(ALL) NOPASSWD:ALL ' | sudo tee /etc/sudoers.d/99-spread-user || true
+
+      ADDRESS "127.0.0.1"
+    discard: |
+      sudo userdel -f -r spread || true
+      sudo rm -f /etc/sudoers.d/99-spread-user
+
+    # HACK: spread does not pass environment variables set on runner
+    # Manually pass specific environment variables
+    environment:
+      CI: '$(HOST: echo $CI)'
+    systems:
+      - ubuntu-22.04:
+          username: spread
+
+suites:
+  tests/spread/mongodb/lxd/:
+    summary: Spread tests for MongoDB VM
+  tests/spread/mongodb/microk8s/:
+    summary: Spread tests for MongoDB Kubernetes
+  tests/spread/mongos/lxd/:
+    summary: Spread tests for Mongos VM
+  tests/spread/mongos/microk8s/:
+    summary: Spread tests for Mongos VM
+
+path: /root/spread_project
+
+kill-timeout: 3h
+environment:
+  PATH: $PATH:$(pipx environment --value PIPX_BIN_DIR)
+  CONCIERGE_JUJU_CHANNEL: 3.6/stable
+prepare: |
+  snap refresh --hold
+  chown -R root:root "$SPREAD_PATH"
+  cd "$SPREAD_PATH"
+  snap install --classic concierge
+
+  concierge prepare --trace
+
+  # Install charmcraft & pipx (on lxd-vm backend)
+  pipx install tox
+  pipx install poetry
```

```suggestion
  # Install charmcraft & pipx (on lxd-vm backend)
  concierge prepare --trace

  pipx install tox poetry
```
for consistency with sql charms